components/automate-ui/src/app/components/resource-dropdown/resource-dropdown.component.ts

```diff
@@ -86,17 +87,19 @@ export class ResourceDropdownComponent implements OnInit, OnChanges {
   }
 
   handleClickOutside(): void {
-    if (this.active) {
+    // Arriving here, the main goal is to close this dropdown if it is open.
+    if (this.dropdownState === 'open') {
       this.closeDropdown();
     }
-    if (this.activating) {
-      this.active = true;
-      this.activating = false;
+    // Now that we have checked the above,
+    // it is safe to complete opening if the click occurred when the dropdown was closed.
+    if (this.dropdownState === 'opening') {
```

Comment:
can you describe further the circumstances when the dropdown is `opening`?

Reply:
The comments in the file explain that if you read them in sequence:
```
// toggleDropdown:
// we cannot go directly to 'open' because handleClickOutside,
// firing next on the same event that arrived here, would then immediately close it.
// => so it is set to 'opening' instead of 'open'. In other words, delay going to 'open'.

// handleClickOutside()
// Arriving here, the main goal is to close this dropdown if it is open.
// check if 'open' -- and now it will not be if coming from toggleDropdown.
. . .
// Now that we have checked the above,
// it is safe to complete opening if the click occurred when the dropdown was closed.
// => so now we set to 'open'
   
If you have more questions happy to try to explain further, but perhaps live chat might be better? Up to you.